doc/tools/coqrst/coqdomain.py

```diff
@@ -1143,6 +1155,33 @@ def GrammarProductionRole(typ, rawtext, text, lineno, inliner, options={}, conte
 
 GrammarProductionRole.role_name = "production"
 
+
+def GlossaryDefRole(typ, rawtext, text, lineno, inliner, options={}, content=[]):
+    """Marks the definition of a glossary term inline in the text.  Matching :term:`XXX`
+    constructs will link to it.  The term will also appear in the Glossary Index.
+
+    Example::
+
+       A :gdef:`prime` number is divisible only by itself and 1.
+    """
+    #pylint: disable=dangerous-default-value, unused-argument
+    env = inliner.document.settings.env
+    std = env.domaindata['std']['objects']
+    key = ('term', text)
+
+    if key in std:
+        MSG = 'Duplicate object: {}; other is at {}'
+        msg = MSG.format(text, env.doc2path(std[key][0]))
```

Comment:
`std[key][0]` is always `term`; is that the right argument to give to doc2path?

Reply:
It is correct, it is the filename.  The std domain dictionary is set up differently from the coq domain dictionary.  This is how I add the entry to std: `std[key] = (env.docname, targetid)`